From: Carsten Schlote <schlote@vahanus.net>
Date: Mon, 10 Mar 2008 22:57:01 +0100
Subject: [PATCH] [strace-4.5.15] Added patches for m68k/v4e

Simple patch to get it compiling.

[m68k] Patch to get strace working with m68k/v4e targets

- Adds some conditional code to leave out stuff on m68k/v4e and get it 
  compiling.

Signed-off-by: Carsten Schlote <schlote@vahanus.net>


Index: strace-4.5.15/signal.c
===================================================================
--- strace-4.5.15.orig/signal.c	2008-03-10 22:50:11.292765846 +0100
+++ strace-4.5.15/signal.c	2008-03-10 22:50:37.794276081 +0100
@@ -129,6 +129,7 @@
 };
 #else /* !I386 */
 #ifdef M68K
+#ifndef __mcoldfire__
 struct sigcontext
 {
 	unsigned long sc_mask;
@@ -141,6 +142,7 @@
 	unsigned long sc_pc;
 	unsigned short sc_formatvec;
 };
+#endif
 #endif /* M68K */
 #endif /* !I386 */
 #endif /* !HAVE_ASM_SIGCONTEXT_H */
Index: strace-4.5.15/system.c
===================================================================
--- strace-4.5.15.orig/system.c	2008-03-10 22:50:11.792794341 +0100
+++ strace-4.5.15/system.c	2008-03-10 22:50:51.795073942 +0100
@@ -71,8 +71,10 @@
 #endif
 
 #ifdef SYS_cacheflush
+#ifndef __mcoldfire__
 #include <asm/cachectl.h>
 #endif
+#endif
 
 #ifdef HAVE_LINUX_USTNAME_H
 #include <linux/utsname.h>
